### PR TITLE
PTW-25: Ensure we always have a getId getter and support for paginati…

### PR DESCRIPTION
**Pull request by peterchallett**
…on settings per entity. Some code refactoring.

### CODE REVIEW

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformColumnExtensions.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        $column = $this;
+        if($column->isPrimary()) {
+            //@TODO - assumes that we only have one primary key column. If multiple then we have an issue...
+            if(strtolower($column->getColumnName()) !== 'id') {
+                $column->writePrimaryKeyFetcher($writer, $nativeType, 'getId');
+            }
+        }
+    }
+
+    public function writePrimaryKeyFetcher(WriterInterface $writer, string $nativeType, string $getterName = 'fetchId') {
```
Comment (TomEdwards) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformColumnExtensions.php:26]: param $getterName could be mandatory

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/Model/Column.php [hunk 2]
```diff
@@ -131,6 +135,8 @@ public function writeGetterAndSetter(WriterInterface $writer)
                 $this->writePrimaryKeyFetcher($writer, $nativeType);
```
Comment (TomEdwards) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/Model/Column.php:135]: is this still required? generates additional field over and above "getId" (called fetchId)

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformFieldAnnotations.php [hunk 3]
```diff
@@ -105,7 +106,7 @@ public function processFields(?string $comment = null): self
     public function generateAnnotationPropertyDetails(string $fieldName, ?string $typeKey = ""): string
     {
         $typeInfo = (object)$this->typeInfo[$typeKey];
-        $details = '"'.$fieldName.'"';
+        $details = '"'.$this->fields[$fieldName]->filterName.'"';
```
Comment (TomEdwards) [inline lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformFieldAnnotations.php:109]: Is this a future thing? Looks like ->filterName and key (name) share the same value

**Review by TomEdwards — CHANGES_REQUESTED** [review]
Looks good, some minor questions

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/Model/Column.php (hunk 2 above)
Comment (peterchallett) [reply lib/MwbExporter/Formatter/Doctrine2/Annotation/Model/Column.php:135]: I've left it in as fetchId was used by Dom in integration of the event store so we'll remove when he can adjust that and switch over to getId vs fetchId.

Comment (TomEdwards) [reply lib/MwbExporter/Formatter/Doctrine2/Annotation/Model/Column.php:135]: Ok - in that case can we adjust to add in the "fetchId" string?

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformFieldAnnotations.php (hunk 3 above)
Comment (peterchallett) [reply lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformFieldAnnotations.php:109]: For search  on fk fields name !== filterName for the @ApiFilter annotation. e.g. offering_family_id => offeringFamily

File: lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformColumnExtensions.php (hunk 1 above)
Comment (peterchallett) [reply lib/MwbExporter/Formatter/Doctrine2/Annotation/OnCorps/ApiPlatformColumnExtensions.php:26]: Will be reworked very soon anyway when we deprecate fetchId

**Pull request merged**